src/nt_users_tool/net_commands.py

```diff
@@ -0,0 +1,72 @@
+from os import popen
+from concurrent.futures import Executor, ThreadPoolExecutor
+
+from typing import NamedTuple
+
+from constants import LAST_RELEVANT_ELEMENT_POSITION
+
+class NTUserInfo(NamedTuple):
+    """Dataclass for a user of the bosch network
+
+    :param NamedTuple: Full name, nt_user and expiration date (fetched from net user /domain)
+    """
+    full_name: str
+    nt_user: str
+    expiration_date: str
+
+def get_nt_user_string(nt_user: str) -> str:
+    """Executes a net command with given nt_user and returns the response as a string.
+
+    :param nt_user: The nt_user that will be used in the net command.
+    :return: The response as a string.
+    """
+    input = nt_user.upper()
+    command = (f"net user /domain {input}")
+    response = popen(command)
+    string_response = response.read()
+    return string_response
+
+def get_all_nt_user_string(list_nt_user: list) -> list:
+    """Performs get_nt_user_string on all elements of list_nt_user.
+
+    :param list_nt_user: The list of nt_user.
+    :return: The list of net command reponse to all elements of list_nt_user.
+    """
+    list_of_nt_user_string = []
+    for nt_user in list_nt_user:
+        list_of_nt_user_string.append(get_nt_user_string(nt_user))
+    return list_of_nt_user_string
+
+def extract_nt_user_info(nt_user_string: str) -> NTUserInfo:
+    """Extracts relevant data from a string generated with a given nt_user.
+
+    :param nt_user_string: Net command response as a string with given nt_user. 
```

Comment:
Based on the description of this param, I would recommand naming it something like "nt_command_response"

Reply:
Fixed in 04d8101.